qutebrowser/browser/downloads.py

```diff
@@ -657,11 +658,42 @@ def _set_filename(self, filename, *, force_overwrite=False,
             self._filename = create_full_filename(self.basename,
                                                   os.path.expanduser('~'))
 
+        dirname = os.path.dirname(self._filename)
+        if not os.path.exists(dirname):
+            txt = ("<b>{}</b> does not exist. Create it?".
+                   format(html.escape(
+                       os.path.join(dirname, ""))))
+            self._ask_create_parent_question("Create directory?", txt,
+                                             force_overwrite,
+                                             remember_directory)
+        else:
+            self._after_create_parent_question(force_overwrite,
+                                               remember_directory)
+
+    def _after_create_parent_question(self,
+                                      force_overwrite, remember_directory):
+        """After asking about parent directory.
+
+        Args:
+            force_overwrite: Force overwriting existing files.
+            remember_directory: If True, remember the directory for future
+                                downloads.
+        """
+        global last_used_directory
+
+        try:
+            os.makedirs(os.path.dirname(self._filename))
+        except OSError as e:
+            # Unlikely, but could be created before
+            # we get a chance to create it.
+            if e.errno != errno.EEXIST:
+                raise
```

Comment:
Since we're using Python 3, you should be able to do:

```python
except FileExistsError:
    pass
```

However, we should really handle any `OSError` here - if e.g. there's no permission to create the directory, you'd get a crash dialog at the moment, right?